### PR TITLE
Limit initial time step to avoid control system deadlocks

### PR DESCRIPTION
**Pull request by wthrowe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ControlSystem/Actions/InitializeMeasurements.hpp [hunk 1]
```diff
@@ -119,6 +129,40 @@ struct InitializeMeasurements {
...
+    for (const auto& fot :
+         Parallel::get<domain::Tags::FunctionsOfTime>(cache)) {
+      earliest_expiration =
+          std::min(earliest_expiration, fot.second->time_bounds()[1]);
+    }
+    const auto& time_step = db::get<::Tags::TimeStep>(box);
+    const auto start_time = time_step.slab().start();
+    if ((start_time + time_step).value() > earliest_expiration) {
+      TimeDelta new_step{};
+      if constexpr (Metavariables::local_time_stepping) {
```
Comment (knelli2) [inline src/ControlSystem/Actions/InitializeMeasurements.hpp]: [optional] You could put the `if constexpr` inside the `db::mutate`. Avoids the intermediate variable.

File: tests/Unit/ControlSystem/Actions/Test_InitializeMeasurements.cpp [hunk 2]
```diff
@@ -266,9 +297,13 @@ void test_initialize_measurements(const bool ab_active, const bool c_active) {
 
 SPECTRE_TEST_CASE("Unit.ControlSystem.InitializeMeasurements",
                   "[ControlSystem][Unit]") {
-  test_initialize_measurements(true, true);
-  test_initialize_measurements(true, false);
-  test_initialize_measurements(false, true);
-  test_initialize_measurements(false, false);
+  test_initialize_measurements<false>(true, true);
+  test_initialize_measurements<false>(true, false);
```
Comment (knelli2) [inline tests/Unit/ControlSystem/Actions/Test_InitializeMeasurements.cpp]: [optional]

```cpp
  for (const auto& [ab_active, c_active] :
       cartesian_product(std::array{true, false}, std::array{true, false})) {
    test_initialize_measurements<false>(ab_active, c_active);
    test_initialize_measurements<true>(ab_active, c_active);
  }
```

**Review by knelli2 — COMMENTED** [review]
In the BBH metavariables, the `InitializeMeasurements` action is at the end of the list, so it's fine that we change the step size there. But what prevents somebody from putting another initialization action after it that again changes the step size to something undesirable by the control system? Not saying this needs to hold up this PR, but something we should think about.

**Comment by wthrowe** [comment]
Initialization actions are only allowed to decrease the time step, otherwise they could destabilize the evolution.  That won't interfere with this code.

The one exception I could imagine is a hypothetical initial run of some StepChoosers to pick a time step, but that's not really any worse than the existing restrictions on the step action ordering.

**Review by knelli2 — APPROVED** [review]
Looks good

**Pull request merged**